mpactpy/core.py

```diff
@@ -0,0 +1,277 @@
+from __future__ import annotations
+from typing import List, Any, Literal, Dict
+from math import isclose
+
+from mpactpy.material import Material
+from mpactpy.pinmesh import PinMesh
+from mpactpy.pin import Pin
+from mpactpy.module import Module
+from mpactpy.lattice import Lattice
+from mpactpy.assembly import Assembly
+from mpactpy.utils import list_to_str, is_rectangular, unique
+
+
+class Core():
+    """ Core of an MPACT model
+
+    Parameters
+    ----------
+    assembly_map : List[List[Assembly]]
+        2-D map of the core assemblies
+    symmetry_opt : SymmetryOption
+        Core symmetry ("360", "90")
+    quarter_sym_opt : QuarterSymmetryOption
+        Quarter core centerline symmetry
+        (i.e. whether the centerline bisects an assembly through the
+        center, or passes between assemblies along the edge)
+
+    Attributes
+    ----------
+    symmetry_opt : SymmetryOption
+        Core symmetry ("360", "90")
+    quarter_sym_opt : QuarterSymmetryOption
+        Quarter core centerline symmetry
+        (i.e. whether the centerline bisects an assembly through the
+        center, or passes between assemblies along the edge)
+    nx : int
+        Number of modules along the x-dimension
+    ny : int
+        Number of modules along the y-dimension
+    nz : int
+        Number of modules along the z-dimension
+    height : float
+        The total height of the core (cm)
+    mod_dim : Assembly.ModDim
+        The x,y,z dimensions of the ray-tracing module
+    assembly_map : List[List[Assembly]]
+        2-D map of the core assemblies
+    assemblies : List[Assembly]
+        The unique assemblies of this core
+    lattices : List[Lattice]
+        The unique lattices of this core
+    modules : List[Module]
+        The unique modules of this core
+    pins : List[Pin]
+        The unique pins of this core
+    pinmeshes : List[PinMesh]
+        The unique pin meshes of this core
+    materials : List[Material]
+        The unqiue materials of this core
+    """
+
+    SymmetryOption = Literal["360", "90", ""]
+    QuarterSymmetryOption = Literal["EDGE", "CENT", ""]
+
+    @property
+    def symmetry_opt(self) -> SymmetryOption:
+        return self._symmetry_opt
+
+    @property
+    def quarter_sym_opt(self) -> QuarterSymmetryOption:
+        return self._quarter_sym_opt
+
+    @property
+    def nx(self) -> int:
+        return len(self.assembly_map)
```

Comment:
A minor thing, but since these values don't change after `__init__`, why not just set them there?
```suggestion
        return self._nx
```

Reply:
Fixed in https://github.com/UT-Computational-NE/MPACTPy/pull/22